### PR TITLE
feat: add job progress logging for cell-by-cell generation

### PR DESCRIPTION
**Pull request by eric-tramel**
## Summary
- Adds detailed progress logging during cell-by-cell column generation in `ColumnWiseDatasetBuilder`
- Extracts progress tracking into a testable `ProgressTracker` class
- Uses thread-safe counters to track completion, success, and failure counts
- Reports progress every ~10% with percentage complete, success/failure counts, processing rate (rec/s), and ETA

## Example output

```
🐙 Processing LLM_TEXT column 'response' with 8 concurrent workers
🧭 LLM_TEXT column 'response' will report progress every 10 record(s).
📈 LLM_TEXT column 'response' progress: 10/100 (10%) complete, 10 ok, 0 failed, 2.45 rec/s, eta 36.7s
📈 LLM_TEXT column 'response' progress: 20/100 (20%) complete, 19 ok, 1 failed, 2.51 rec/s, eta 31.9s
📈 LLM_TEXT column 'response' progress: 30/100 (30%) complete, 28 ok, 2 failed, 2.48 rec/s, eta 28.2s
...
📈 LLM_TEXT column 'response' progress: 100/100 (100%) complete, 97 ok, 3 failed, 2.52 rec/s, eta 0.0s
```

## Test plan
- [x] Run dataset generation with cell-by-cell columns and verify progress logs appear
- [x] Verify progress updates at ~10% intervals
- [x] Test with varying batch sizes to confirm interval calculation works correctly
- [x] Unit tests for `ProgressTracker` class (19 tests covering thread safety, edge cases, logging)

### CODE REVIEW

File: packages/data-designer-engine/src/data_designer/engine/dataset_builders/utils/progress_tracker.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    def _log_progress_unlocked(self) -> None:
+        """Log current progress. Must be called while holding the lock."""
+        elapsed = time.perf_counter() - self.start_time
+        rate = self.completed / elapsed if elapsed > 0 else 0.0
+        remaining = max(0, self.total_records - self.completed)
+        eta = f"{(remaining / rate):.1f}s" if rate > 0 else "unknown"
+        percent = (self.completed / self.total_records) * 100 if self.total_records else 100.0
+
+        logger.info(
+            "📈 %s progress: %d/%d (%.0f%%) complete, %d ok, %d failed, %.2f rec/s, eta %s",
```
Comment (johnnygreco) [inline packages/data-designer-engine/src/data_designer/engine/dataset_builders/utils/progress_tracker.py]: super small nit, but this way it will match how we format the other logs
```suggestion
        logger.info(
            "  |-- 📈 %s progress: %d/%d (%.0f%%) complete, %d ok, %d failed, %.2f rec/s, eta %s",
```

**Review by johnnygreco — CHANGES_REQUESTED** [review]
Love pulling it out into a class like this! Thanks @eric-tramel! 

Just the smallest of small nits
